src/actions/posts.js

```diff
@@ -227,22 +227,18 @@ export function createPost(post, files = []) {
                             id: pendingPostId,
                             failed: true,
                         };
-
-                        const actions = [
-                            {type: PostTypes.CREATE_POST_FAILURE, error},
-                        ];
+                        dispatch({type: PostTypes.CREATE_POST_FAILURE, error});
 
                         // If the failure was because: the root post was deleted or
                         // TownSquareIsReadOnly=true then remove the post
                         if (error.server_error_id === 'api.post.create_post.root_id.app_error' ||
-                            error.server_error_id === 'api.post.create_post.town_square_read_only'
+                            error.server_error_id === 'api.post.create_post.town_square_read_only' ||
+                            error.server_error_id === 'plugin.message_will_be_posted.dismiss_post'
                         ) {
-                            actions.push(removePost(data));
+                            dispatch(removePost(data));
```

Comment:
I don't fully understand why the batched actions prevented the post from being removed -- did we never try sending removePost before this?

Reply:
Not with a batch, I don't think. There's in issue open for redux-batch-actions here https://github.com/tshelburne/redux-batched-actions/issues/3